web/daemon/server.go

```diff
@@ -815,6 +820,116 @@ func extract(c *gin.Context) {
 	}
 }
 
+// @Summary Create backup
+// @Description Creates a full backup of the server
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Router /api/servers/{id}/backup/create [post]
+// @Security OAuth2Application[server.backup.create]
+func createBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	isRunning, err := server.IsRunning()
+	if err != nil {
+		response.HandleError(c, err, http.StatusInternalServerError)
+		return
+	} else if isRunning {
+		response.HandleError(c, pufferpanel.ErrBackupServerRunning, http.StatusBadRequest)
+		return
+	}
+
+	backupFileName, fileSize, err := server.CreateBackup()
+
+	pufferError, ispufferError := err.(*pufferpanel.Error)
+	if ispufferError && pufferpanel.ErrBackupInProgress.Is(pufferError) {
+		response.HandleError(c, err, http.StatusBadRequest)
+	} else if response.HandleError(c, err, http.StatusInternalServerError) {
+	} else {
+		c.JSON(http.StatusOK, &pufferpanel.ServerBackupResponse{BackupFileName: backupFileName, FileSize: fileSize})
+	}
+}
+
+// @Summary Delete backup
+// @Description Delete a backup of the server
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Param fileName query string true "File Name"
+// @Router /api/servers/{id}/backup/delete [post]
+// @Security OAuth2Application[server.backup.delete]
+func deleteBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	fileName := c.Query("fileName")
+
+	err := server.DeleteBackup(fileName)
+
+	pufferError, ispufferError := err.(*pufferpanel.Error)
+	if ispufferError && pufferpanel.ErrBackupInProgress.Is(pufferError) {
+		response.HandleError(c, err, http.StatusBadRequest)
+	} else if response.HandleError(c, err, http.StatusInternalServerError) {
+	} else {
+		c.Status(http.StatusOK)
+	}
+}
+
+// @Summary Restore backup
+// @Description Restore a full backup of the server
+// @Success 204 {object} nil
+// @Param id path string true "Server ID"
+// @Param fileName query string true "File Name"
+// @Router /api/servers/{id}/backup/restore [post]
+// @Security OAuth2Application[server.backup.restore]
+func restoreBackup(c *gin.Context) {
+	server := getServerFromGin(c)
+	fileName := c.Query("fileName")
+	isRunning, err := server.IsRunning()
+	if err != nil {
+		response.HandleError(c, err, http.StatusInternalServerError)
+		return
+	} else if isRunning {
+		response.HandleError(c, pufferpanel.ErrBackupServerRunning, http.StatusBadRequest)
+		return
+	}
+
+	err = server.RestoreBackup(fileName)
+
+	pufferError, ispufferError := err.(*pufferpanel.Error)
+	if ispufferError && pufferpanel.ErrBackupInProgress.Is(pufferError) {
+		response.HandleError(c, err, http.StatusBadRequest)
+	} else if response.HandleError(c, err, http.StatusInternalServerError) {
+	} else {
+		c.Status(http.StatusOK)
```

Comment:
StatusNoContent

Reply:
fixed in [3afa64c](https://github.com/pufferpanel/pufferpanel/pull/1379/commits/3afa64cabbf2a66b294418fec744d28d5024bb52)